tests/tpp_backend_setup.py

```diff
@@ -25,6 +26,25 @@
 Base = declarative_base()
 
 
+# Hack: in order to improve the variety of our test data, this mixin intercepts object
+# initialization, adding a time part to any kwarg that is for a datetime field that has
+# been provided as a date.
+#
+# (We cannot add this behaviour by subclassing Base, because subclasses of Base are
+# expected to have a __tablename__ attribute.)
```

Comment:
You can use the `cls` argument to `declarative_base()` to set the base class:
https://docs.sqlalchemy.org/en/13/orm/extensions/declarative/api.html#sqlalchemy.ext.declarative.declarative_base